src/mamba.ts

```diff
@@ -18,15 +14,47 @@ export function sign(urlString: string, privateKey: string) {
   const url = new URL(urlString);
   const { hash } = url;
   url.hash = "";
-  url.searchParams.set("ts", Date.now().toString());
+  url.searchParams.set(
+    "ts",
+    Math.round(new Date().getTime() / 1000).toString()
+  );
   const sign = crypto.createSign("SHA512");
   sign.update(url.toString());
 
-  /* TODO: The sign() function will throw exceptions if the key isn't formatted right.
-   * We should handle all the possible exceptions and return better error messages.
-   */
-  url.searchParams.set("sig", sign.sign(privateKey, "base64"));
-  url.hash = hash;
+  try {
+    const sig = sign
+      .sign(privateKey)
+      .toString("base64")
+      .replace(/\+/g, "-")
+      .replace(/\//g, "_");
```

Comment:
Do you need to drop the `=` padding here as well?

Reply:
For some reason, no. Just encoding that to `%3D` is fine afaik.

Reply:
(which is done automatically via `searchParams.set`)